net/rpc/server.go

```diff
@@ -534,6 +550,14 @@ func (server *Server) readRequest(codec ServerCodec) (service *service, mtype *m
 		return
 	}
 
+	if server.preBodyInterceptor != nil {
+		// Allow interceptor to halt servicing of the request
+		err = server.preBodyInterceptor(req.ServiceMethod, req.SourceAddr)
+		if err != nil {
+			return
```

Comment:
Do we need to surface the error somehow to return the right error to the client?

Reply:
`err` is a named return value of type `error` on L542 and gets returned implicitly with a lone `return` statement.